server/modules/data-access/cachedSingleAccountDynamoTable.js

```diff
@@ -0,0 +1,74 @@
+/* Copyright (c) Trainline Limited, 2016-2017. All rights reserved. See LICENSE.txt in the project root for license information. */
+
+'use strict';
+
+let mkArn = require('modules/data-access/dynamoTableArn').mkArn;
+let attachAuditMetadata = require('modules/data-access/dynamoAudit').attachAuditMetadata;
+let describeDynamoTable = require('modules/data-access/describeDynamoTable');
+let hashKeyAttributeName = require('modules/data-access/dynamoTableDescription').hashKeyAttributeName;
+let tableArn = require('modules/data-access/dynamoTableDescription').tableArn;
+let dynamoVersion = require('modules/data-access/dynamoVersion');
+let dynamoTableCache = require('modules/data-access/dynamoTableCache');
+let dynamoSoftDelete = require('modules/data-access/dynamoSoftDelete');
+let fp = require('lodash/fp');
+
+function constructor(physicalTableName, { ttl }) {
+  let cachedTable = dynamoTableCache(physicalTableName, { ttl });
+  let tableDescriptionPromise = () => mkArn({ tableName: physicalTableName }).then(describeDynamoTable);
+
+  function create(item) {
+    return tableDescriptionPromise().then(description =>
+      fp.flow(
+        attachAuditMetadata,
+        record => ({ record }),
+        dynamoVersion.compareAndSetVersionOnCreate(hashKeyAttributeName(description)),
+        cachedTable.create.bind(null, tableArn(description))
+      )(item)
+    );
+  }
+
+  function $delete(item, expectedVersion) {
+    return tableDescriptionPromise().then((description) => {
```

Comment:
Just an arbitrary prefix to avoid collision with the `delete` keyword. I believe Fabio has used this convention elsewhere.